test/unit/tasks/seeds_test.rb

```diff
@@ -103,27 +110,32 @@ def seed
   end
 
   test "does update template that was not modified by user" do
-    seed
+    seed('020-provisioning_templates_list.rb', '070-provisioning_templates.rb')
     ProvisioningTemplate.without_auditing { ProvisioningTemplate.unscoped.find_by_name('Kickstart default').update(:template => 'test') }
-    seed
-    refute_equal ProvisioningTemplate.unscoped.find_by_name('Kickstart default').template, 'test'
+    seed('020-provisioning_templates_list.rb', '070-provisioning_templates.rb')
+    assert_not_equal ProvisioningTemplate.unscoped.find_by_name('Kickstart default').template, 'test'
   end
 
   test "doesn't add a template back that was deleted" do
-    seed
-    assert_equal 1, ProvisioningTemplate.unscoped.
-      where(:name => 'Kickstart default').destroy_all.size
-    seed
-    refute ProvisioningTemplate.unscoped.find_by_name('Kickstart default')
+    seed('020-provisioning_templates_list.rb', '070-provisioning_templates.rb')
+    with_auditing(ProvisioningTemplate) do
+      assert_equal 1, ProvisioningTemplate.unscoped.where(:name => 'Kickstart default').destroy_all.size
+    end
+    assert SeedHelper.audit_modified?(ProvisioningTemplate, 'Kickstart default')
+    seed('020-provisioning_templates_list.rb', '070-provisioning_templates.rb')
+    assert_not ProvisioningTemplate.unscoped.find_by_name('Kickstart default')
   end
 
   test "doesn't add a template back that was renamed" do
-    seed
-    tmpl = ProvisioningTemplate.unscoped.find_by_name('Kickstart default')
-    tmpl.name = 'test'
-    tmpl.save!
-    seed
-    refute ProvisioningTemplate.unscoped.find_by_name('Kickstart default')
+    seed('020-provisioning_templates_list.rb', '070-provisioning_templates.rb')
+    with_auditing(ProvisioningTemplate) do
+      tmpl = ProvisioningTemplate.unscoped.find_by_name('Kickstart default')
+      tmpl.name = 'test'
+      tmpl.save!
+    end
```

Comment:
This is the key thing that fixes random failures - ensures that auditing is enabled during this test.